src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -1,29 +1,91 @@
+import org.broadinstitute.consent.http.enumeration.AssociationType;
+import org.broadinstitute.consent.http.models.Consent;
 import org.broadinstitute.consent.http.models.DataSet;
 import org.broadinstitute.consent.http.models.DataSetProperty;
+import org.broadinstitute.consent.http.models.DataUse;
 import org.broadinstitute.consent.http.models.Dictionary;
 import org.broadinstitute.consent.http.models.dto.DataSetDTO;
 import org.broadinstitute.consent.http.models.dto.DataSetPropertyDTO;
 
 public class DatasetService {
 
+    public static final String DATASET_NAME_KEY = "Dataset Name";
+    public static final String CONSENT_NAME_PREFIX = "DUOS-DS-CG-";
+    private final ConsentDAO consentDAO;
     private final DataSetDAO dataSetDAO;
 
     @Inject
-    public DatasetService(DataSetDAO dataSetDAO) {
+    public DatasetService(ConsentDAO consentDAO, DataSetDAO dataSetDAO) {
+        this.consentDAO = consentDAO;
         this.dataSetDAO = dataSetDAO;
     }
 
+    /**
+     * Create a minimal consent from the data provided in a Dataset.
+     *
+     * @param dataset The DataSetDTO
+     * @return The created Consent
+     */
+    public Consent createConsentForDataset(DataSetDTO dataset) {
+        String consentId = UUID.randomUUID().toString();
+        Optional<DataSetPropertyDTO> nameProp = dataset.getProperties()
+            .stream()
+            .filter(p -> p.getPropertyName().equalsIgnoreCase(DATASET_NAME_KEY))
+            .findFirst();
+        // Typically, this is a construct from ORSP consisting of dataset name and some form of investigator code.
+        // In our world, we'll use that dataset name if provided, or the alias.
+        String groupName = nameProp.isPresent() ? nameProp.get().getPropertyValue() : dataset.getAlias();
+        String name = CONSENT_NAME_PREFIX + dataset.getDataSetId();
+        Date createDate = new Date();
+        if (Objects.nonNull(dataset.getDataUse())) {
+            boolean manualReview = isConsentDataUseManualReview(dataset.getDataUse());
+            /*
+             * Consents created for a dataset do not need the following properties:
+             * use restriction
+             * data user letter
+             * data user letter name
+             * translated use restriction
+             */
+            consentDAO.insertConsent(consentId, manualReview, null, dataset.getDataUse().toString(),
+                  null, name, null, createDate, createDate, null,
+                  true, groupName, dataset.getDacId());
+            String associationType = AssociationType.SAMPLESET.getValue();
+            consentDAO.insertConsentAssociation(consentId, associationType, dataset.getDacId());
+            return consentDAO.findConsentById(consentId);
+        } else {
+            throw new NullPointerException("Dataset is missing Data Use information. Consent could not be created.");
+        }
+
+    }
+
+    private boolean isConsentDataUseManualReview(DataUse dataUse) {
```

Comment:
I was having a bit of trouble understanding what to do with this, as it appears that many of the fields in DataUse won't actually have values, so it wouldn't be safe to arbitrarily access them. In v2, we can "fix" this by guaranteeing that all these fields are defaulted to false on the UI on the info coming in, but that would be a departure from how things are done today.

Reply:
I think that's fine. The code checks for null-ness so we shouldn't fail in the case you're describing. "Manual Review"-ness is purely a function of checking a variety of fields for any value at all.